compiler/circle-interpreter-test/src/circle-interpreter.test.cpp

```diff
@@ -0,0 +1,263 @@
+
+  delete[] buffer1;
+  delete[] buffer2;
+  return true;
+}
+
+TEST_F(circle_interpreter_test, show_help_msg)
+{
+  if (!initialize())
+  {
+    FAIL();
+    return;
+  }
+
+  std::string command = _circle_interpreter_path + " -h";
+  if (!run(command))
+  {
+    FAIL();
+    return;
+  }
+
+  const auto pos = _result.find("Usage: ./circle-interpreter");
+  ASSERT_NE(std::string::npos, pos);
+}
+
+TEST_F(circle_interpreter_test, valid_command)
+{
+  if (!initialize())
+  {
+    FAIL();
+    return;
+  }
+
+  std::string model = _artifacts_path + "/Conv2D_000.circle";
+  std::string input_prefix = _artifacts_path + "/Conv2D_000.circle.input";
+  std::string output_prefix = "/tmp/Conv2D_000.circle.output";
+  std::string generated_output = output_prefix + "0";
+  std::remove(generated_output.c_str());
+  std::string command =
+    _circle_interpreter_path + " " + model + " " + input_prefix + " " + output_prefix;
+  if (!run(command))
+  {
+    FAIL();
+    return;
+  }
+
+  std::string expected_output = _artifacts_path + "/Conv2D_000.circle.output0";
+
+  if (!compare(generated_output, expected_output))
+  {
+    FAIL();
+    return;
+  }
+}
+
+TEST_F(circle_interpreter_test, invalid_option_NEG)
+{
+  if (!initialize())
+  {
+    FAIL();
+    return;
+  }
+
+  std::string model = _artifacts_path + "/Conv2D_000.circle";
+  std::string command = _circle_interpreter_path + " " + model;
+  if (!run(command))
+  {
+    FAIL();
+    return;
+  }
+
+  const auto pos = _result.find("Invalid argument");
+  ASSERT_NE(std::string::npos, pos);
+}
+
+TEST_F(circle_interpreter_test, not_existing_model_NEG)
+{
+  if (!initialize())
+  {
+    FAIL();
+    return;
+  }
+
+  std::string not_existing_model = _artifacts_path + "/non_exist_file.foo";
+  std::string input_prefix = _artifacts_path + "/Conv2D_000.circle.input";
+  std::string output_prefix = "/tmp/Conv2D_000.circle.output";
+  std::remove(output_prefix.c_str());
+  std::string command =
+    _circle_interpreter_path + " " + not_existing_model + " " + input_prefix + " " + output_prefix;
+  if (!run(command))
+  {
+    FAIL();
+    return;
+  }
+
+  const auto pos = _result.find("Failed to load");
+  ASSERT_NE(std::string::npos, pos);
+}
+
+TEST_F(circle_interpreter_test, invalid_input_prefix_NEG)
+{
+  if (!initialize())
+  {
+    FAIL();
+    return;
+  }
+
+  std::string model = _artifacts_path + "/Conv2D_000.circle";
+  std::string input_prefix = _artifacts_path + "/non_exist_filee.foo";
+  std::string output_prefix = "/tmp/Conv2D_000.circle.output";
+  std::remove(output_prefix.c_str());
+  std::string command =
+    _circle_interpreter_path + " " + model + " " + input_prefix + " " + output_prefix;
+  if (!run(command))
+  {
+    FAIL();
+    return;
+  }
+
+  const auto pos = _result.find("Cannot open file");
+  ASSERT_NE(std::string::npos, pos);
+}
+
+// Case of given input data has less number of values than model's input
+TEST_F(circle_interpreter_test, input_data_mismatch_NEG)
```

Comment:
I worried how `circle-interpreter` will work if the given input data's number of values mismatch with input tensor size. 

1) If circle model's input size is smaller than the input data's size, execution succeed.
2) If circle model's input size is larger than the input data's size, execution failed with following message.
ERROR: Failed to read data from file "${INPUT_FILE_PATH}".

So, I added this negative test. Is this behavior (succeed for larger size input data) intended? 
/cc @jinevening @seanshpark

Reply:
> Is this behavior (succeed for larger size input data) intended?

IMHO, it would be better to generate an error (ex: throw an exception) in `circle-interpreter` when the size of input data mismatches with the size of the model input. There may be a way to get the size of input file in cpp code.

So, it would be better to remove this test, because error message may not be the same with the current one.